### PR TITLE
Create function code action for lhs conditionals

### LINKED ISSUE #36321: Create function code action doesn't work in lhs expression in ternary conditional expression
Consider the following case
```
function testConditional3(int arg) {
    string res = isGreaterThanZero(arg) ? "a" : "b";
}
```
currently create function code action creates following method while we expect a `boolean` return
```
function isGreaterThanZero(int i) returns string {
    return "";
}
```

_Originally posted by @gayalkuruppu in https://github.com/ballerina-platform/ballerina-lang/pull/35815#pullrequestreview-981097638_

### PR DESCRIPTION
**Pull request by bimsarapathiraja**
Fixes #36321

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java [hunk 1]
```diff
@@ -424,6 +427,12 @@ public void visit(WhileStatementNode whileStatementNode) {
 
     @Override
     public void visit(ConditionalExpressionNode conditionalExpressionNode) {
+        if (this.functionCallExpressionNode != null &&
+                PositionUtil.isWithinLineRange(this.functionCallExpressionNode.lineRange(),
+                conditionalExpressionNode.lhsExpression().lineRange())) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java:432]: Let's add the `!conditionalExpressionNode.lhsExpression().isMissing()` condition as well

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java [hunk 2]
```diff
@@ -424,6 +427,12 @@ public void visit(WhileStatementNode whileStatementNode) {
...
+            checkAndSetTypeDescResult(TypeDescKind.BOOLEAN);
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java:433]: Now that we have introduced the TypesAPI, we can use `checkAndSetTypeResult(semanticModel.get().types().BOOLEAN)`

File: language-server/modules/langserver-core/src/test/resources/command/create-function/config/create_function_in_conditional_expression.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                "newText": "\n\nfunction isGreaterThanZero(int i) returns boolean {\n    return false;\n}"
+              }
+            ]
+          }
+        ]
+      }
+    },
+    "jsonrpc": "2.0"
+  }
+}
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/test/resources/command/create-function/config/create_function_in_conditional_expression.json:41]: Let's add a new line

File: language-server/modules/langserver-core/src/test/resources/command/create-function/source/create_function_in_conditional_expression.bal [hunk 4]
```diff
@@ -0,0 +1,3 @@
+function testConditional3(int arg) {
+    string res = isGreaterThanZero(arg) ? "a" : "b";
+}
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/test/resources/command/create-function/source/create_function_in_conditional_expression.bal:3]: new line

**Review by malinthar — CHANGES_REQUESTED** [review]
Changes look good! Let's address the following minor comments.

**Comment by mohanvive** [comment]
@bimsarapathiraja Shall we address the review comments and the conflicts.

**Review by gayalkuruppu — APPROVED** [review]
LGTM

**Comment by mohanvive** [comment]
> Changes look good! Let's address the following minor comments.

@malinthar Need your review and approval to merge this.

**Review by mohanvive — APPROVED** [review]

**Pull request merged**